chain/client/src/client.rs

```diff
@@ -2233,18 +2230,14 @@ impl Client {
             return Ok(ProcessTxResponse::InvalidTx(err));
         }
 
-        let shard_id = account_id_to_shard_id(
-            self.epoch_manager.as_ref(),
-            tx.transaction.signer_id(),
-            &epoch_id,
-        )?;
+        let shard_id = shard_layout.account_id_to_shard_id(tx.transaction.signer_id());
```

Comment:
nit
```
let shard_uid = shard_layout.account_id_to_shard_uid(...);
let shard_id = shard_uid.shard_id(); 
```
And then you can get rid of the line getting shard uid below. Totally optional, it's just stylistic.